regression_tests/utils.go

```diff
@@ -0,0 +1,277 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"regexp"
+	"strings"
+
+	walletclient "github.com/bitcoin-sv/spv-wallet-go-client"
+	"github.com/bitcoin-sv/spv-wallet-go-client/xpriv"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/joho/godotenv"
+)
+
+const (
+	atSign                   = "@"
+	domainPrefix             = "http://"
+	domainSuffixSharedConfig = "/v1/shared-config"
+	spvWalletIndexResponse   = "Welcome to the SPV Wallet ✌(◕‿-)✌"
+
+	ClientOneURLEnvVar         = "CLIENT_ONE_URL"
+	ClientTwoURLEnvVar         = "CLIENT_TWO_URL"
+	ClientOneLeaderXPrivEnvVar = "CLIENT_ONE_LEADER_XPRIV"
+	ClientTwoLeaderXPrivEnvVar = "CLIENT_TWO_LEADER_XPRIV"
+)
+
+var explicitHTTPURLRegex = regexp.MustCompile(`^https?://`)
+
+type User struct {
+	XPriv   string `json:"xpriv"`
+	XPub    string `json:"xpub"`
+	Paymail string `json:"paymail"`
+}
+
+type Config struct {
+	ClientOneURL         string
+	ClientTwoURL         string
+	ClientOneLeaderXPriv string
+	ClientTwoLeaderXPriv string
+}
+
+type WalletResponse struct {
+	Message string `json:"message"`
+}
+
+func SaveConfig(config *Config) error {
```

Comment:
please make those methods and structs internal if possible

Reply:
changed to internal